keel-core-test/src/main/kotlin/com/netflix/spinnaker/keel/persistence/CombinedRepositoryTests.kt

```diff
@@ -408,6 +410,48 @@ abstract class CombinedRepositoryTests<D : DeliveryConfigRepository, R : Resourc
           expectThat(subject.getDeliveryConfigForApplication("keel").name).isEqualTo(configName)
         }
       }
+
+      context("submitting delivery config with invalid environment name as a constraint") {
+        val submittedConfig = SubmittedDeliveryConfig(
+          name = configName,
+          application = "keel",
+          serviceAccount = "keel@spinnaker",
+          artifacts = setOf(DockerArtifact(name = "org/thing-1", deliveryConfigName = configName, reference = "thing")),
+          environments = setOf(
+            SubmittedEnvironment(
+              name = "test",
+              resources = setOf(
+                SubmittedResource(
+                  metadata = mapOf("serviceAccount" to "keel@spinnaker"),
+                  kind = TEST_API_V1.qualify("whatever"),
+                  spec = DummyResourceSpec(data = "o hai")
+                )
+              ),
+              constraints = emptySet()
+            ),
+              SubmittedEnvironment(
+              name = "test",
+            resources = setOf(
+              SubmittedResource(
+                metadata = mapOf("serviceAccount" to "keel@spinnaker"),
+                kind = TEST_API_V1.qualify("whatever"),
+                spec = DummyResourceSpec(data = "o hai")
+              )
+            ),
+            constraints = setOf(DependsOnConstraint(environment = "notRealEnvironment"))
+          )
+          )
+        )
+
+        test("an error is thrown and config is not persisted") {
+          expectCatching {
+            subject.upsertDeliveryConfig(submittedConfig)
+          }.failed()
+            .isA<MissingEnvironmentReferenceException>()
+
+          expectThat(subject.allResourceNames().size).isEqualTo(0)
```

Comment:
if you remove the resources like suggested above you can just check to make sure the delivery config is not persisted.

Reply:
I couldn't find the right statement for making sure the delivery config table is empty (I saw this check is used for the same thing, so I used it).. can you point me to the right statement?

Reply:
I would just do:
```
expectCatching {
            subject.getDeliveryConfig(name)
          }.failed()
            .isA<NoSuchDeliveryConfigException>()
```

Reply:
oooo right! thank you!